### ISSUE TITLE
Add support for Decimal type

### ISSUE THREAD

**Issue by mamcx** [rusqlite/rusqlite#1059]
Similar to [rust postgres](https://github.com/paupino/rust-decimal/blob/master/src/postgres.rs), it will be great if this lib add support for the[ Decimal type](https://docs.rs/rust_decimal/latest/rust_decimal/).

My case is that I'm building my own mini-orm, and differently to rust postgres, I can't implement ToSql/FromSql for it:

```rust
impl FromSql for Decimal {
    fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
        unimplemented!()
    }
}

```

```bash
error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
   --> db/src/sqlite.rs:220:1
    |
220 | impl FromSql for Decimal {
    | ^^^^^^^^^^^^^^^^^-------
    | |                |
    | |                `rust_decimal::Decimal` is not defined in the current crate
    | impl doesn't use only types from inside the current crate
    |
    = note: define and implement a trait or new type instead
```

**Comment by gwenn** [comment]
The `orphan rule` means that only `rust_decimal` or `rusqlite` crates can implement `FromSql`/`ToSql` for `Decimal` directly.

**Comment by mamcx** [comment]
Oh, I see. I ask there. Thanks.

**Comment by mamcx** [comment]
Ok, so can I put a pull request for this here?

**Comment by gwenn** [comment]
Yep, something like https://github.com/rusqlite/rusqlite/pull/506

**Comment by mamcx** [comment]
Making the test I found is not possible to round-trip [Decimal::Max](https://docs.rs/rust_decimal/latest/rust_decimal/struct.Decimal.html#associatedconstant.MAX), and it fails with `InvalidColumnType` instead of report is an overflow.

In [get<I: RowIndex, T: FromSql>(https://github.com/rusqlite/rusqlite/blob/master/src/row.rs#L280)  it returns this error.

Can I return the overflow instead or let it be as-is and you look at it later?

**Comment by gwenn** [comment]
You should be able to use [FromSqlError::Other](https://docs.rs/rusqlite/latest/rusqlite/types/enum.FromSqlError.html#variant.Other), no ?

**Comment by mamcx** [comment]
Yeah, but is still returned as `InvalidType`, no matter the original error...

**Comment by mamcx** [comment]
Ok, I send the pull request, tell me if I need to improve it anyway!
